base/show.jl

```diff
@@ -2718,7 +2718,7 @@ function type_depth_limit(str::String, n::Int; maxdepth = nothing)
         end
         prev = di
     end
-    return String(take!(output))
+    return unsafe_takestring!(output)
```

Comment:
Why is this guaranteed to be safe?

Reply:
Because there is no way for this function to load a reference to `output`.